src/libos/src/net/socket/unix_socket/stream/socket.rs

```diff
@@ -0,0 +1,190 @@
+        })
+    }
+
+    pub fn path(&self) -> UnixAddr {
+        self.path.read().unwrap().clone()
+    }
+
+    pub fn end(&self) -> &SgxMutex<Option<Arc<UnixEnd>>> {
+        &self.end
+    }
+
+    pub fn peer_name(&self) -> Result<UnixAddr> {
+        if let Some(end) = &*self.end.lock().unwrap() {
+            Ok(end.peer_name())
+        } else {
+            return_errno!(ENOTCONN, "the socket is not connected");
+        }
+    }
+
+    pub fn is_connected(&self) -> bool {
+        self.end.lock().unwrap().is_some()
+    }
+
+    pub fn is_blocking(&self) -> bool {
+        self.is_blocking.load(Ordering::SeqCst)
+    }
+
+    pub fn set_non_blocking(&self) {
+        self.is_blocking.store(false, Ordering::SeqCst);
+        let end = self.end.lock().unwrap();
+        end.as_ref().map(|c| c.set_non_blocking());
+    }
+
+    pub fn set_blocking(&self) {
+        self.is_blocking.store(true, Ordering::SeqCst);
+        let end = self.end.lock().unwrap();
+        end.as_ref().map(|c| c.set_blocking());
+    }
+
+    // TODO: create the corresponding file in the fs
+    pub fn bind(&self, addr: &UnixAddr) -> Result<()> {
+        if !self.path().is_unnamed() {
+            return_errno!(EINVAL, "the socket is already bound");
+        }
+
+        STREAM_SPACE.add_binder(addr)?;
+
+        *self.path.write().unwrap() = addr.clone();
+        if let Some(ref end) = *self.end.lock().unwrap() {
+            end.set_name(&addr);
+        }
+        Ok(())
+    }
+
+    //TODO: add backlog support
+    pub fn listen(&self, backlog: i32) -> Result<()> {
+        if self.path().is_unnamed() {
+            return_errno!(EINVAL, "the socket is not bound");
+        }
+
+        STREAM_SPACE.add_listener(&self.path())
+    }
+
+    /// A non-blocking accept
+    pub fn accept(&self, flags: FileFlags) -> Result<(Self, Option<UnixAddr>)> {
+        if self.path().is_unnamed() {
+            return_errno!(EINVAL, "the socket is not bound");
+        }
+
+        let sock = STREAM_SPACE.pop_incoming(&self.path())?;
+
+        if flags.contains(FileFlags::SOCK_NONBLOCK) {
+            sock.set_non_blocking();
+        }
+
+        let addr = sock.peer_name().ok();
+
+        debug!("the accepted socket is {:?} from {:?}", sock, addr);
+
+        Ok((sock, addr))
+    }
+
+    /// Backlog is temparily not supported so connect will not block
+    pub fn connect(&self, addr: &Option<UnixAddr>) -> Result<()> {
+        debug!("connect to {:?}", addr);
+
+        if addr.is_none() || addr.contains(&UnixAddr::UnNamed) {
+            return_errno!(EINVAL, "invalid address");
+        }
+
+        if self.is_connected() {
+            return_errno!(EISCONN, "already connected");
+        }
+
+        let path = addr.as_ref().unwrap();
+        if *path == self.path() {
+            return_errno!(EINVAL, "self connect is not supported");
+        }
+
+        let (end_self, end_incoming) = UnixEnd::new_duplex_channel()?;
+        end_incoming.set_name(path);
```

Comment:
In this single function, one single concept has been called by three different names: addr, path, and name. This entire PR needs to be consistent about the naming.

Reply:
Good point. It leaves inconsistency where code is changed too many times.